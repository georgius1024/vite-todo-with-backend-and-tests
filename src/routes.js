import Todos from "./views/Todos.vue";
import Todo from "./views/Todo.vue";
import TodoNew from "./views/TodoNew.vue";

export default [
  { path: "/new", component: TodoNew },
  { path: "/:id", component: Todo },
  { path: "/", component: Todos },
];
