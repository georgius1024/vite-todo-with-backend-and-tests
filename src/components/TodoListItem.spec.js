import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import component from "./TodoListItem.vue";
import { createRouter, createWebHistory } from "vue-router";
import routes from "../routes";

const router = createRouter({
  history: createWebHistory(),
  routes,
});

describe("TodoListItem", () => {
  beforeEach(async () => {
    router.push("/");
    await router.isReady();
  });

  const propsData = {
    todo: {
      id: 101,
      title: "hello",
      completed: true,
    },
  };

  const global = {
    plugins: [router],
  };

  it("mounts without error", () => {
    const wrapper = mount(component, { propsData, global });
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("item click navigates", async () => {
    const push = vi.spyOn(router, 'push');
    const wrapper = mount(component, { propsData, global });
    await wrapper.find("a").trigger("click");
    expect(push).toBeCalledWith('/101');
  });
});
