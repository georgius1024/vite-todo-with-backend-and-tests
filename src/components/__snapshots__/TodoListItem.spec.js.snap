// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoListItem > mounts without error 1`] = `"<a href=\\"/101\\" class=\\"list-group-item list-group-item-action\\">âœ“<span class=\\"ms-1\\">hello</span></a>"`;
